src/ck-core/ckfutures.C

```diff
@@ -262,6 +262,31 @@ void *CkWaitFutureID(CkFutureID handle)
   return value;
 }
 
+std::pair<void*, CkFutureID> CkWaitAnyID(const std::vector<CkFutureID>& handles) {
+  auto fs = &(CpvAccess(futurestate));
+  const auto ready = std::any_of(handles.begin(), handles.end(),
+    [&fs](const CkFutureID& id) { return fs->is_ready(id); });
+  if (!ready) fs->request(handles, std::make_shared<FutureToThread>(CthSelf()));
+  do {
+    if (!ready) CthSuspend();
```

Comment:
What is responsible for resuming this thread?

Reply:
The FutureBOC object wakes it up when it fulfills the `FutureToThread` request.

Reply:
Note, a `FutureToThread` request is also used on ckfutures.h:251 as well -- so it may be better to put some design notes about how future requests near their definition. I will put a back-reference to those sections here.

Reply:
Comment added (see updates)